jobs/e2e-test-cycloud.groovy

```diff
@@ -0,0 +1,230 @@
+REVISION_DESCRIPTION = params.REVISION_DESCRIPTION ?: params.CYPRESS_GIT_REVISION;
+E2E_URL = params.URL[-1] == '/' ? params.URL.substring(0, params.URL.length() - 1): params.URL;
+
+currentBuild.displayName = ("${currentBuild.displayName} " +
+                            "(${REVISION_DESCRIPTION})");
+
+// We use the build name as a unique identifier for user notifications.
+BUILD_NAME = "build e2e-cypress-test #${env.BUILD_NUMBER} (${E2E_URL}: ${params.REVISION_DESCRIPTION})"
+
+// GIT_SHA1 is the sha1 for CYPRESS_GIT_REVISION.
+GIT_SHA1 = null;
+
+// We have a dedicated set of workers for the second smoke test.
+WORKER_TYPE = (params.USE_FIRSTINQUEUE_WORKERS
+               ? 'ka-firstinqueue-ec2' : 'ka-test-ec2');
+
+def initializeGlobals() {
+   NUM_WORKER_MACHINES = params.NUM_WORKER_MACHINES.toInteger();
+}
+
+
+def _setupWebapp() {
+   GIT_SHA1 = kaGit.resolveCommitish("git@github.com:Khan/webapp",
+                                        params.CYPRESS_GIT_REVISION);
+
+   kaGit.safeSyncToOrigin("git@github.com:Khan/webapp", GIT_SHA1);
+
+   dir("webapp/services/static") {
+      sh("yarn install --frozen-lockfile");
+   }
+}
+
+// Run all the test-clients on all the worker machine, in parallel.
+def runAllTestClients() {
+   // We want to swallow any framework exceptions unless *all* the
+   // clients have raised a framework exception.  Our theory is that
+   // if one client dies unexpectedly the others can compensate, but
+   // if they all do, then there's nothing more we can do.
+   def onException = {
+      echo("Worker raised an exception");
+      WORKERS_RAISING_EXCEPTIONS++;
+      if (WORKERS_RAISING_EXCEPTIONS == NUM_WORKER_MACHINES) {
+         echo("All worker machines failed!");
+         throw new TestFailed("All worker machines failed!");
+      }
+   }
+
+   def jobs = [:];
+   for (i = 0; i < NUM_WORKER_MACHINES; i++) {
+      def workerId = i;  // avoid scoping problems
+      jobs["e2e-test-${workerId}"] = {
+         swallowExceptions({
+            onWorker(WORKER_TYPE, '2h') {
+                _setupWebapp();
+                runE2ETests(workerId);
+            }
+         }, onException);
+      };
+   }
+   parallel(jobs);
+}
+
+
+def runTests() {
+   try {
+     runAllTestClients();
+    } catch (TestsAreDone e) {
+      // Ignore this "error": it's thrown on successful test completion.
+      echo("Tests are done!");
+    }
+}
+
+
+def runE2ETests(workerId) {
+   echo("Starting e2e tests for worker ${workerId}");
+
+   // Determine which environment we're running against, so we can provide a tag
+   // in the LambdaTest build.
+   def e2eEnv = E2E_URL == "https://www.khanacademy.org" ? "prod" : "preprod";
+
+   // NOTE: We hard-code the values here as it is an experiment that will be
+   // removed soon.
+   def runE2ETestsArgs = ["CYPRESS_PROJECT_ID=2c1iwj && CYPRESS_RECORD_KEY=4c530cf3-79e5-44b5-aedb-f6f017f38cb5 && ./dev/cypress/e2e/tools/start-cypress-cloud-run.ts"];
```

Comment:
```suggestion
   def runE2ETestsArgs = ["env", "CYPRESS_PROJECT_ID=2c1iwj", "CYPRESS_RECORD_KEY=4c530cf3-79e5-44b5-aedb-f6f017f38cb5", "./dev/cypress/e2e/tools/start-cypress-cloud-run.ts"];
```

Reply:
What you have now won't work, it will try to find a binary with the name of that entire string.

An alternative to using `env` would be to set the environment in jenkins via `withEnv`.